Duplicati/Library/Backend/OneDrive/MicrosoftGraphBackend.cs

```diff
@@ -353,89 +356,95 @@ public async Task PutAsync(string remotename, Stream stream, CancellationToken c
                 // The documentation seems somewhat contradictory - it states that uploads must be done sequentially,
                 // but also states that the nextExpectedRanges value returned may indicate multiple ranges...
                 // For now, this plays it safe and does a sequential upload.
-                HttpRequestMessage createSessionRequest = new HttpRequestMessage(HttpMethod.Post, string.Format("{0}/root:{1}{2}:/createUploadSession", this.DrivePrefix, this.RootPath, NormalizeSlashes(remotename)));
-
-                // Indicate that we want to replace any existing content with this new data we're uploading
-                this.PrepareContent(new UploadSession() { Item = new DriveItem() { ConflictBehavior = ConflictBehavior.Replace } });
-
-                HttpResponseMessage createSessionResponse = await m_client.SendAsync(createSessionRequest, cancelToken);
-                UploadSession uploadSession = this.ParseResponse<UploadSession>(createSessionResponse);
-
-                // If the stream's total length is less than the chosen fragment size, then we should make the buffer only as large as the stream.
-                int bufferSize = (int)Math.Min(this.fragmentSize, stream.Length);
-
-                byte[] fragmentBuffer = new byte[bufferSize];
-                int read = 0;
-                for (int offset = 0; offset < stream.Length; offset += read)
+                using (HttpRequestMessage createSessionRequest = new HttpRequestMessage(HttpMethod.Post, string.Format("{0}/root:{1}{2}:/createUploadSession", this.DrivePrefix, this.RootPath, NormalizeSlashes(remotename))))
+                /* Indicate that we want to replace any existing content with this new data we're uploading */
+                using (createSessionRequest.Content = this.PrepareContent(new UploadSession() { Item = new DriveItem() { ConflictBehavior = ConflictBehavior.Replace } }))

```

Comment:
This was forgotten before I guess as you mentioned in your PR.

Reply:
That's what it seems like (which is particularly embarrassing as I was the one to originally write this code). I'm also not sure why I missed all the disposing then..